### ISSUE TITLE
Typing

### ISSUE THREAD

**Issue by jmsmkn** [DIAGNijmegen/rse-gcapi#166]
We tried to type this at some point but ran into trouble. See:

- https://github.com/DIAGNijmegen/rse-gcapi/pull/150/commits/e5b6f47d0548a7795bce4d3d7f6c800eb4015a96
- https://github.com/DIAGNijmegen/rse-roadmap/issues/213

**Comment by chrisvanrun** [comment]
Fixing this on the go as the hybrid constructs are being removed.

It does have me wondering: should we make mypy stricter? Setting `strict=True`, logically results in load of errors but there might be a balance somewhere with these, a subset of strict:

```
no_implicit_optional = True
warn_unused_ignores = True
warn_return_any = True
warn_redundant_casts = True
warn_unused_configs = True
strict_equality = True
```

T.b.h. I think the typing is a great aid but it should not become a hassle in a small code base that is gcapi. I vote we add typing where it makes sense.

**Comment by jmsmkn** [comment]
All detailed in https://mypy.readthedocs.io/en/stable/existing_code.html

TLDR: Turn each of the features on one at a time, fixing the errors as you go, until you [complete the strict set](https://mypy.readthedocs.io/en/stable/existing_code.html#introduce-stricter-options) and can set that.
